beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -240,23 +257,29 @@ func (vs *Server) ProposeBeaconBlock(ctx context.Context, req *ethpb.GenericSign
 		return nil, fmt.Errorf("could not broadcast block: %v", err)
 	}
 
+	var scs []*ethpb.SignedBlobSidecar
 	if blk.Version() >= version.Deneb {
-		b, ok := req.GetBlock().(*ethpb.GenericSignedBeaconBlock_Deneb)
-		if !ok {
-			return nil, status.Error(codes.Internal, "Could not cast block to Deneb")
-		}
-		if len(b.Deneb.Blobs) > fieldparams.MaxBlobsPerBlock {
-			return nil, status.Errorf(codes.InvalidArgument, "Too many blobs in block: %d", len(b.Deneb.Blobs))
+		if wasBlinded {
```

Comment:
I feel like there is a code smell around this, a lot of different arrays used, can't put my finger on how to clean it up just yet but some parts can be broken into its own function or cleaned up some way just for readability.

Reply:
slightly better: 2c8530e79